hypothesis-python/src/hypothesis/strategies/_internal/datetime.py

```diff
@@ -62,30 +88,50 @@ def draw_capped_multipart(data, min_value, max_value):
         result[name] = val
         cap_low = cap_low and val == low
         cap_high = cap_high and val == high
+    if hasattr(min_value, "fold"):
```

Comment:
Not sure if it's worth it, but since `dateutil.tz` has a backport of the `fold` attribute, when using a `dateutil` time zone, it may make sense to use `tz.enfold` in Python 3.5.

If you want to do that, you may want to just unconditionally set `result["fold"]`, and strip it out before applying it to the datetime or time.

I don't think `tz.enfold` works on `datetime.time`, though fold is pretty meaningless for `time` anyway.

Reply:
Given the extra complexity of version-and-optional-package-specific logic, I don't think it's worth it.  Python 3.5 is only ~five months to EOL and ~3% of our downloads anyway...